controllers/agents/svc/serviceclass_controller.go

```diff
@@ -106,15 +112,24 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	// then, write all the registered services up to the primaza cluster
 	if serviceClass.DeletionTimestamp.IsZero() {
-		handler := func(remote_client client.Client, rs v1alpha1.RegisteredService) error {
-			op, err := controllerutil.CreateOrUpdate(ctx, remote_client, &rs, func() error { return nil })
+		handler := func(remote_client client.Client, rs v1alpha1.RegisteredService, secret *v1.Secret) error {
+			clone := &v1alpha1.RegisteredService{ObjectMeta: *rs.ObjectMeta.DeepCopy()}
+			op, err := controllerutil.CreateOrUpdate(ctx, remote_client, clone, func() error {
+				*clone = rs
+				return nil
+			})
 			if err != nil {
 				reconcileLog.Error(err, "Failed to create registered service",
 					"service", rs.Name,
 					"namespace", rs.Namespace)
 			} else {
 				reconcileLog.Info("Wrote registered service", "service", rs.Name, "namespace", rs.Namespace, "operation", op)
 			}
+
+			if secret != nil {
+				_, err1 := controllerutil.CreateOrUpdate(ctx, remote_client, secret, func() error { return controllerutil.SetOwnerReference(clone, secret, remote_client.Scheme()) })
+				err = errors.Join(err, err1)
```

Comment:
We seem to be preparing the error List on L299, so we can avoid this operation here right?

Reply:
I've changed the implementation to aggregate errors there.